test/libsolidity/syntaxTests/storageLayoutSpecifier/type_uint_max.sol

```diff
@@ -1,3 +1,3 @@
 contract at layout at type(uint).max { }
 // ----
-// TypeError 6396: (22-36): The base slot of the storage layout must evaluate to a rational number.
+// TypeError 1505: (22-36): The base slot expression cannot be evaluated during compilation.
```

Comment:
I was hoping this would work already in this PR, but apparently it's another thing that requires proper compile-time evaluation... https://github.com/ethereum/solidity/issues/11183#issuecomment-1246835929

Reply:
Well, one thing that should work after the PR are simple constants initialized with expressions that we evaluate in unlimited precision. Please make sure that this is the case.

Reply:
It is. We should still have at least one dedicated test whose goal is specifically to show that constants work.

Reply:
I think you're also missing tests with some expression on constants or with constants that refer to other constants. These should all work.